### PR TITLE
use document-uri instead of source-file

### PR DESCRIPTION
**Pull request by labkey-matthewb**
#### Rationale
document-uri is better for avoiding duplicates

#### Related Pull Requests
* 

#### Changes
*

### CODE REVIEW

File: core/src/org/labkey/core/admin/AdminController.java [hunk 1]
```diff
@@ -10952,10 +10952,16 @@ public Object execute(SimpleApiJsonForm form, BindException errors) throws Excep
...
+                if (cspReport != null)
+                {
+                    String urlString = cspReport.getString("document-uri");
+                    if (urlString != null)
+                    {
+                        String path = new URLHelper(urlString).deleteParameters().getPath();
+                        if (null == reports.put(path, Boolean.TRUE))
+                            _log.warn("ContentSecurityPolicy warning on page: " + urlString + "\n" + jsonStr);
+                    }
+                }
```
Comment (labkey-tchad) [inline core/src/org/labkey/core/admin/AdminController.java]: Should use `JSONObject.opt*` for items that might not be there; `get*` will just throw an exception if they don't find anything. Also, `JSONObject.toString` can take an indent factor to make it look pretty.
```suggestion
                JSONObject cspReport = jsonObj.optJSONObject("csp-report");
                if (cspReport != null)
                {
                    String urlString = cspReport.optString("document-uri", null);
                    if (urlString != null)
                    {
                        String path = new URLHelper(urlString).deleteParameters().getPath();
                        if (null == reports.put(path, Boolean.TRUE))
                        {
                            var jsonStr = jsonObj.toString(2);
                            _log.warn("ContentSecurityPolicy warning on page: " + urlString + "\n" + jsonStr);
                        }
                    }
                }
```

**Review by labkey-tchad — CHANGES_REQUESTED** [review]
I thought of another way to 'disable' this endpoint. It could no-op if the logger is turned off. Like:
```
if (!_log.isWarnEnabled())
{
    return new JSONObject().put("success", false);
}
``` 

Also, we should update `server/webapps/log4j2.xml` to put these warning into a separate file. That would make it possible for tests to check for new warnings (and fail if we want).

File: core/src/org/labkey/core/admin/AdminController.java [hunk 2]
```diff
@@ -10946,18 +10945,30 @@ public class ContentSecurityPolicyReportAction extends ReadOnlyApiAction<SimpleA
...
                 var jsonStr = jsonObj.toString();
-                JSONObject cspReport = jsonObj.getJSONObject("csp-report");
-                String urlString = cspReport.getString("source-file");
-                String path = new URLHelper(urlString).deleteParameters().getPath();
-                if (null == reports.put(path,Boolean.TRUE))
-                    _log.warn("ContentSecurityPolicy warning:\n" + jsonStr);
+                JSONObject cspReport = jsonObj.optJSONObject("csp-report");
+                if (cspReport != null)
+                {
+                    String urlString = cspReport.optString("document-uri");
```
Comment (labkey-tchad) [inline core/src/org/labkey/core/admin/AdminController.java]: Unless specified, `optString(key)`'s default value is `""`, not `null`.
```suggestion
                    String urlString = cspReport.optString("document-uri", null);
```

`org.json` has numerous questionable design decisions with regard to `null` values.

**Review by labkey-tchad — APPROVED** [review]
One small fix

File: core/src/org/labkey/core/admin/AdminController.java (hunk 2 above)
Comment (labkey-matthewb) [reply core/src/org/labkey/core/admin/AdminController.java]: Thanks, now I know.

**Pull request merged**